java/org/apache/catalina/valves/AbstractAccessLogValve.java

```diff
@@ -1756,10 +1776,14 @@ protected AccessLogElement createAccessLogElement(String name, char pattern) {
                 return new DateAndTimeElement(name);
             case 'T':
                 // ms for milliseconds, us for microseconds, and s for seconds
-                if ("ms".equals(name)) {
-                    return new ElapsedTimeElement(false, true);
+                if ("ns".equals(name)) {
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.MILLISECONDS);
                 } else if ("us".equals(name)) {
-                    return new ElapsedTimeElement(true, false);
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.MICROSECONDS);
+                } else if ("ms".equals(name)) {
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.MILLISECONDS);
+                } else if ("fs".equals(name)) {
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.SECONDS_FRACTIONAL);
```

Comment:
Strictly, that is fs is femto seconds which doesn't look right. Maybe sf?

Reply:
Good question. I actually thought of femtoseconds and then thought "who actually knows about those?" and now I guess I have my answer.

We could use `sf`, sure... also `s.` which is a little obscure. There is no reason is has to be two-letters, either. `fracsec`?

Reply:
It is the electronics engineer in me - capacitors have really small values.

`fracsec` works for me.

Reply:
Done/